### ISSUE TITLE
bigip_bigiq_as3 Resource doesn't honor bigiq_token_auth setting

### ISSUE THREAD

**Issue by dan-da-man-86** [F5Networks/terraform-provider-bigip#436]
When using Terraform to connect to BIG-IQ via the BIG-IP provider (resource "bigip_bigiq_as3"), the option for bigiq_token_auth is available, but not honored.  When token auth is enabled for the resource and Basic auth is disabled in BIG-IQ via bash: set-basic-auth off (BIG-IQ default behavior), I get the following error:
----------------------------------------------------
Error: HTTP 401 :: <html>
<head><title>401 Authorization Required</title></head>
<body bgcolor="white">
<center><h1>401 Authorization Required</h1></center>
<hr><center>webd</center>
</body>
</html>

  on bigip_as3.tf line 74, in resource "bigip_bigiq_as3" "deploy_as3_to_bigips":
  74: resource "bigip_bigiq_as3" "deploy_as3_to_bigips" {
-------------------------------------------------------

After running bash: set-basic-auth on on BIG-IQ, the resource is able to successfully deploy the AS3 through BIG-IQ.

Token auth in general is tested as working on BIG-IQ with the same credentials used here via the BIG-IP provider in TF as well as with Postman. The resource uses the same credential variables as the BIG-IP provider that successfully uses Token auth.

Resource config:
resource "bigip_bigiq_as3" "deploy_as3_to_bigips" {
  for_each          = local.f5_vm_names
  bigiq_address     = var.bigIqHost
  bigiq_user        = var.bigIqUsername
  bigiq_password    = var.bigIqPassword
  bigiq_token_auth  = true
  bigiq_login_ref   = "tmos" #have also tested with value of "local" as well as omitting this setting
  as3_json          = data.template_file.bigip_as3_template[each.key].rendered 
} 

Provider config:
provider "bigip" {
  address         = var.bigIqHost
  username        = var.bigIqUsername
  password        = var.bigIqPassword
  token_auth      = true
  login_ref       = "tmos"
}

BIG-IP Provider v1.7 - also tested on v1.6
Terraform v14.7

Also, the TF provider documention appears to be missing from the following site for "bigiq_token_auth" and "bigiq_login_ref":
https://registry.terraform.io/providers/F5Networks/bigip/latest/docs/resources/bigip_bigiq_as3

**Comment by nmenant** [comment]
tracking this issue with INFRAANO-250

**Comment by dan-da-man-86** [comment]
Update: Tested on updated versions and issue persists.
Terraform 14.9
BIG-IP Provider 1.8.0

**Comment by RavinderReddyF5** [comment]
@dan-da-man-86 can you please set `bigiq_login_ref` to `local` or dont set ant value to it, because it sets to `local` by default  and retry it?

```
resource "bigip_bigiq_as3" "deploy_as3_to_bigips" {
for_each = local.f5_vm_names
bigiq_address = var.bigIqHost
bigiq_user = var.bigIqUsername
bigiq_password = var.bigIqPassword
bigiq_token_auth = true
bigiq_login_ref = "local" #have also tested with value of "local" as well as omitting this setting
as3_json = data.template_file.bigip_as3_template[each.key].rendered
}
```

**Comment by RavinderReddyF5** [comment]
@dan-da-man-86  fixed in v1.9.0, please verify
